packages/cozy-harvest-lib/src/models/ConnectionFlow.spec.js

```diff
@@ -571,64 +501,91 @@ describe('ConnectionFlow', () => {
         Q('io.cozy.files').getById(
           fixtures.createdTriggerWithFolder.attributes.message.folder_to_save
         )
       )
     })
+    delete window.cozy
+    delete window.ReactNativeWebView
+  })
 
-    it('should call the launcher when needed', async () => {
-      const { client } = setup()
-      const flow = new ConnectionFlow(
-        client,
-        fixtures.existingTrigger,
-        fixtures.clientKonnector
-      )
-      window.cozy = {
-        ClientConnectorLauncher: 'react-native'
+  describe('custom konnector policy', () => {
+    const bankingKonnector = {
+      ...fixtures.konnector,
+      partnership: {
+        domain: 'budget-insight.com'
       }
-      window.ReactNativeWebView = {
-        postMessage: jest.fn()
+    }
+
+    const bankingAccount = {
+      ...fixtures.existingAccount,
+      ...fixtures.bankingKonnectorAccountAttributes
+    }
+
+    const onAccountCreationResult = {
+      auth: {
+        bi: {
+          connId: 7
+        }
       }
-      ensureTrigger.mockImplementation(async () => fixtures.existingTrigger)
-      prepareTriggerAccount.mockResolvedValue(fixtures.updatedAccount)
+    }
 
-      await flow.ensureTriggerAndLaunch(client, {
-        account: fixtures.existingAccount,
-        trigger: fixtures.existingTrigger,
-        konnector: fixtures.konnector
+    it('should use custom konnector policy', async () => {
+      saveAccount.mockReset().mockImplementation((konnector, acc) => ({
+        ...acc,
+        _id: fixtures.updatedAccount._id
+      }))
+
+      biKonnectorPolicy.onAccountCreation
+        .mockReset()
+        .mockReturnValue(onAccountCreationResult)
+
+      const { flow, client } = setup()
+
+      jest.spyOn(flow, 'launch')
+
+      await setupSubmit(flow, {
+        konnector: bankingKonnector,
+        account: bankingAccount,
+        userCredentials: fixtures.bankingKonnectorAccountAttributes.auth
       })
-      expect(flow.account).toEqual(fixtures.updatedAccount)
-      expect(window.ReactNativeWebView.postMessage).toHaveBeenCalledWith(
-        JSON.stringify({
-          message: 'startLauncher',
-          value: {
-            connector: fixtures.clientKonnector,
-            account: fixtures.updatedAccount,
-            trigger: fixtures.existingTrigger,
-            job: fixtures.launchedJob
-          }
-        })
+
+      expect(biKonnectorPolicy.onAccountCreation).toHaveBeenCalledTimes(1)
+      expect(ensureTrigger).toHaveBeenCalledTimes(1)
+      expect(flow.launch).toHaveBeenCalledTimes(1)
+      expect(saveAccount).toHaveBeenCalledWith(
+        client,
+        flow.konnector,
+        onAccountCreationResult
       )
     })
-    delete window.cozy
-    delete window.ReactNativeWebView
+  })
+})
+
+describe('ensureTrigger', () => {
+  beforeAll(() => {
+    jest.spyOn(cronHelpers, 'fromFrequency').mockReturnValue('0 0 0 * * 0')
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
```

Comment:
What is the goal of those lines?

Reply:
These lines are not actually needed anymore